.github/workflows/deploy-website.yml

```diff
@@ -24,8 +24,11 @@ jobs:
 
       - name: Install mkdocs
         run: |
-          pip install mkdocs-material mkdocs-minify-plugin mkdocs-redirects \
-                      mkdocs-glightbox mdx-gh-links
+          pip install mkdocs-material==9.0.2 \
+                      mkdocs-minify-plugin==0.6.2 --use-pep517 \
+                      mkdocs-redirects==1.2.0 --use-pep517 \
+                      mkdocs-glightbox==0.3.1 \
+                      mdx-gh-links==0.3
```

Comment:
Is the idea that we'll march these  forward manually at some interval (with the benefit being that we can still generate the doc site against old commits, because they will use old versions that were working at that time)? If so, makes sense!

Reply:
Yes, that's the idea. I'm a big fan of pinned dependency versions as it makes builds repeatable and non-surprising.

Few things are worse than trying to get out an urgent patch, and suddenly you cannot build the project anymore because some of the libs have been updated. Or worse, you can build the project, but the updates introduce subtle bugs...

This has bitten me more than once at work; since then, I pin down everything, and only do controlled upgrades, when I want it, not when I'm forced to.